### DISCUSSION TITLE
How to access Shadecn Form Item value in real time before submitting?

### DISCUSSION THREAD

**Discussion by 2019011821** [shadcn-ui/ui#3216] [Q&A]
So I have been playing with shadecn form based upon react hook form.

Everything is working well except I need a new functionality where I need to save user's current form setting.

It is done so that user can prefilled those value if needed next time.

On form submitting the default operation kicks in. However this is not needed here. I just need the value for saving in the database.
```
<Form {...form} >
<form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
<CommonArticleParts control={form.control} articleMode={articleMode} setShowOutlineBuilder={setShowOutlineBuilder} />
<CreateArticleButton isLoading={isLoading} showOutlineBuilder={showOutlineBuilder} />
</form>
</Form>

```
I tried using sethook but it is slowing down the whole page as the form is big.

Any productive approach here?
Thanks

**Comment by aidrecabrera** [comment]
> I need to save user's current form setting

Could you please provide more details on what you mean by `"save user's current form setting"`? Are you thinking of saving that state locally, or were you considering sending it to a server for storage?

Another thing, when you mention prefilling those values next time around, are you thinking of automatically populating the form fields with the saved data when the user returns to the form?

It would be great if you provide an rough example.
